src/qibolab/transpilers/placer.py

```diff
@@ -0,0 +1,260 @@
+import random
+
+import networkx as nx
+from qibo.config import log, raise_error
+from qibo.models import Circuit
+
+from qibolab.transpilers.abstract import Placer, create_circuit_repr
+
+
+def assert_placement(circuit: Circuit, layout: dict, verbose=False) -> bool:
+    """Checks if layout is correct and matches the number of qubits of the circuit.
+
+    Args:
+        circuit (qibo.models.Circuit): Circuit model to check.
+        layout (dict): physical to logical qubit mapping.
+        verbose (bool): If ``True`` it prints info messages.
+
+    Returns ``True`` if the following conditions are satisfied:
+        - layout is written in the correct form.
+        - layout matches the number of qubits in the circuit.
+    otherwise returns ``False``.
+    """
+    if not assert_mapping_consistency(layout, verbose=verbose):
+        return False
+    if circuit.nqubits == len(layout):
+        if verbose:
+            log.info("Layout can be used on circuit.")
+        return True
+    if circuit.nqubits > len(layout):
+        if verbose:
+            log.info("Layout can't be used on circuit. The circuit requires more qubits.")
+        return False
+    if verbose:
+        log.info("Layout can't be used on circuit. Ancillary extra qubits need to be added to the circuit.")
+    return False
+
+
+def assert_mapping_consistency(layout, verbose=False):
+    """Checks if layout is correct.
+
+    Args:
+        layout (dict): physical to logical qubit mapping.
+        verbose (bool): If ``True`` it prints info messages.
+
+    Returns: ``True`` if layout is written in the correct form.
+    otherwise returns ``False``.
+    """
+    values = sorted(layout.values())
+    keys = list(layout.keys())
+    ref_keys = ["q" + str(i) for i in range(len(keys))]
+    if keys != ref_keys:
+        if verbose:
+            log.info("Some physical qubits in the layout may be missing or duplicated")
+        return False
+    if values != list(range(len(values))):
+        if verbose:
+            log.info("Some logical qubits in the layout may be missing or duplicated")
+        return False
+    return True
+
+
+class Trivial(Placer):
+    """Place qubits trivially, same logical and physical placement
+
+    Attributes:
+        connectivity (networkx.Graph): chip connectivity.
+    """
+
+    def __init__(self, connectivity=None):
+        """Args:
+        connectivity (networkx.graph): chip connectivity.
+        """
+        self.connectivity = connectivity
+
+    def __call__(self, circuit: Circuit):
+        """Find the trivial placement for the circuit.
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit model to check.
+        """
+        return dict(zip(list("q" + str(i) for i in range(circuit.nqubits)), range(circuit.nqubits)))
+
+
+class Custom(Placer):
```

Comment:
I would keep `Custom`: it is clear that is a placer, since it is inside the `placer.py` module.

We did the same transition for the transpiler, in order to avoid repetitions:
https://github.com/qiboteam/qibolab/commit/4bd61e9865ef8484761d0d9ee30416c0a1ae9b16

Reply:
In case of ambiguity, at import time you can import with:
```py
from qibolab.transpilers import placer
# placer.Custom
from qibolab.transpilers.placer import Custom as CustomPlacer
# CustomPlacer
```